File: traffic-jam/traffic_jam_test.py
import unittest

import pydantic
import traffic_jam


@pydantic.dataclasses.dataclass
class TestCase:
    intervals: list[int]
    threshold: int
    want: int


class TestTrafficJam(unittest.TestCase):
    def test_calculate(self):
        test_cases = [
            TestCase(intervals=[5, 6, 25, 4], threshold=10, want=15),
            TestCase(intervals=[30, 10, 40], threshold=30, want=40),
        ]
        for test_case in test_cases:
            got = traffic_jam.calculate(test_case.intervals, test_case.threshold)
            self.assertEqual(test_case.want, got)


if __name__ == "__main__":
    unittest.main()

File: traffic-jam/trafficjam_test.go
package main

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

type testCase struct {
	intervals []int
	threshold int
	want      int
}

func TestCalculate(t *testing.T) {
	testCases := []testCase{
		{[]int{5, 6, 25, 4}, 10, 15},
		{[]int{30, 10, 40}, 30, 40},
	}
	calculator := new()
	for _, tc := range testCases {
		got := calculator.Calculate(tc.intervals, tc.threshold)
		assert.Equal(t, tc.want, got)
	}
}
